### PR TITLE
issue-3775 : Allow configuration override of default kinesis endpoints

### PR DESCRIPTION
**Pull request by arihantsurana**
This change addresses the issue - https://github.com/snowplow/snowplow/issues/3775
:sparkles:
This commit enables an optional configuration in config.hocon for both 'scala-stream-collector' and 'stream-enrich' for overriding Kinesis endpoints.
Allowing an optional override on the calculated aws endpoints is useful while using local setup eg. localstack.

### CODE REVIEW

**Comment by arihantsurana** [comment]
I am getting 2/4 build failures on Travis. The errors on these steps indicate environmental issues rather than test failure due to the code changes in this PR:
https://travis-ci.org/snowplow/snowplow/jobs/385498036#L1188
https://travis-ci.org/snowplow/snowplow/jobs/385498039#L1306

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://github.com/snowplow/snowplow/wiki/CLA to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

File: 2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala [hunk 1]
```diff
@@ -74,11 +74,15 @@ package model {
...
   ) extends SinkConfig {
-    val endpoint = region match {
-      case cn@"cn-north-1" => s"https://kinesis.$cn.amazonaws.com.cn"
-      case _ => s"https://kinesis.$region.amazonaws.com"
+    val endpoint = customEndpoint match {
+      case Some(overrideEndpoint) => overrideEndpoint
+      case None => region match {
+        case cn@"cn-north-1" => s"https://kinesis.$cn.amazonaws.com.cn"
+        case _ => s"https://kinesis.$region.amazonaws.com"
+      }
```
Comment (BenFradet) [inline 2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala]: you should be able to do:

```scala
customEndpoint.getOrElse(region match {
  // ...
})
```

same thing below

File: 2-collectors/scala-stream-collector/examples/config.hocon.sample [hunk 2]
```diff
@@ -98,6 +98,10 @@ collector {
       # Region where the streams are located
       region = {{kinesisRegion}}
 
+      ## Endpoint url configuration to override aws kinesis endpoints,
```
Comment (BenFradet) [inline 2-collectors/scala-stream-collector/examples/config.hocon.sample]: I'd say `Optional endpoint url...`, same thing below

**Comment by arihantsurana** [comment]
I signed it!

File: 2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala (hunk 1 above)
Comment (arihantsurana) [reply 2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala]: Done for both.

File: 2-collectors/scala-stream-collector/examples/config.hocon.sample (hunk 2 above)
Comment (arihantsurana) [reply 2-collectors/scala-stream-collector/examples/config.hocon.sample]: Done for both.

**Comment by arihantsurana** [comment]
@BenFradet rearranged commits into two clean ones.

**Comment by snowplowcla** [comment]
@arihantsurana has signed the Software Grant and Corporate Contributor License Agreement

**Comment by BenFradet** [comment]
Thanks a lot, merged :+1:

**Pull request merged**